### ISSUE TITLE
Seg fault in mixed mode (DPPQDCEventIterator)

### ISSUE THREAD

**Issue by fmessi** [tblum/jadaq#5]
Running the VX1740D_31 on the MB machine in mixed mode and the config below sometimes results in a segfault (release build) or a "Assertion failed" (debug), typically after only a few seconds of run time. The MB detector is connected. This is running on local branch `mb_testing_env` based on the devel branch with Hanno's recent output file name changes included (on commit c6317cd). 

This is the backtrace and error message produced:
```
[mb@mbdesktop test10fm]$ gdb --args /home/mb/jadaq/build/jadaq -t 10 -H Config.ini
GNU gdb (GDB) Red Hat Enterprise Linux 8.0.1-36.el7
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/mb/jadaq/build/jadaq...done.
(gdb) r
Starting program: /home/mb/jadaq/build/jadaq -t 10 -H Config.ini
Missing separate debuginfos, use: debuginfo-install glibc-2.17-196.el7_4.2.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Setup 1 digitizer(s):
	VX1740D_31
Starting run 00015 (ID 14ef3dc5dd54788b) 
Start acquisition on digitizer VX1740D_31
[New Thread 0x7fffe72d1700 (LWP 15715)]
Running acquisition loop - Ctrl-C to interrupt
Warning: dropping 8 data points from HDF5 file.
jadaq: /home/mb/jadaq/EventIterator.hpp:88: void DPPQDCEventIterator::GroupIterator::nextGroup(): Assertion `(size - 2) % elementSize == 0' failed.

Thread 1 "jadaq" received signal SIGABRT, Aborted.
0x00007ffff117c1f7 in raise () from /lib64/libc.so.6
Missing separate debuginfos, use: debuginfo-install boost-atomic-1.53.0-27.el7.x86_64 boost-chrono-1.53.0-27.el7.x86_64 boost-date-time-1.53.0-27.el7.x86_64 boost-filesystem-1.53.0-27.el7.x86_64 boost-program-options-1.53.0-27.el7.x86_64 boost-system-1.53.0-27.el7.x86_64 boost-thread-1.53.0-27.el7.x86_64 hdf5-1.8.12-10.el7.x86_64 libaec-1.0.2-1.el7.x86_64 libgcc-4.8.5-16.el7_4.1.x86_64 libstdc++-4.8.5-16.el7_4.1.x86_64 zlib-1.2.7-17.el7.x86_64
(gdb) bt
#0  0x00007ffff117c1f7 in raise () from /lib64/libc.so.6
#1  0x00007ffff117d8e8 in abort () from /lib64/libc.so.6
#2  0x00007ffff1175266 in __assert_fail_base () from /lib64/libc.so.6
#3  0x00007ffff1175312 in __assert_fail () from /lib64/libc.so.6
#4  0x0000000000532385 in DPPQDCEventIterator::GroupIterator::nextGroup (this=0x7fffffffce48) at /home/mb/jadaq/EventIterator.hpp:88
#5  0x000000000053240b in DPPQDCEventIterator::GroupIterator::operator++ (this=0x7fffffffce48) at /home/mb/jadaq/EventIterator.hpp:109
#6  0x00000000005325e7 in DPPQDCEventIterator::operator++ (this=0x7fffffffce30) at /home/mb/jadaq/EventIterator.hpp:166
#7  0x0000000000535654 in DataHandler::Implementation<Data::WaveformElement<Data::ListElement422> >::operator() (this=0x7ded40, eventIterator=...) at /home/mb/jadaq/DataHandler.hpp:157
#8  0x0000000000532a5c in DataHandler::operator() (this=0x7d9cf0, it=...) at /home/mb/jadaq/DataHandler.hpp:45
#9  0x00000000005305be in Digitizer::acquisition (this=0x7d9cd0) at /home/mb/jadaq/Digitizer.cpp:406
#10 0x00000000004d22a5 in main (argc=5, argv=0x7fffffffde38) at /home/mb/jadaq/jadaq.cpp:254
```

[jadaq-run-00015.txt](https://github.com/ess-dmsc/jadaq/files/2181147/jadaq-run-00015.txt)

**Comment by fmessi** [comment]
To have the data in mixed mode with the detector, only one group of wires and two groups of strips were acquired... 
if not, the acquisition crashes with error "CAEN_DGTZ_OutOfMemory"

**Comment by tblum** [comment]
Fixed by commit 3e4d7b15af84c98929260c329647f1f83f9825ac
